econml/sklearn_extensions/linear_model.py

```diff
@@ -1675,53 +1675,76 @@ class StatsModelsLinearRegression(_StatsModelsWrapper):
         The statsmodels-style fit arguments; keys can include 'cov_type'
     """
 
-    def __init__(self, fit_intercept=True, cov_type=None):
+    def __init__(self, fit_intercept=True, cov_type="HC0"):
         self.cov_type = cov_type
         self.fit_intercept = fit_intercept
         return
 
-    def _check_input(self, X, y, sample_weight, sample_var):
+    def _check_input(self, X, y, sample_weight, freq_weight, sample_var):
         """Check dimensions and other assertions."""
+
+        if X is None:
+            X = np.empty((y.shape[0], 0))
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+
+        # set default values for None
         if sample_weight is None:
             sample_weight = np.ones(y.shape[0])
-        elif np.any(np.not_equal(np.mod(sample_weight, 1), 0)):
-            raise AttributeError("Sample weights must all be integers for inference to be valid!")
-
+        if freq_weight is None:
+            freq_weight = np.ones(y.shape[0])
         if sample_var is None:
-            if np.any(np.not_equal(sample_weight, 1)):
-                warnings.warn(
-                    "No variance information was given for samples with sample_weight not equal to 1, "
-                    "that represent summaries of multiple original samples. Inference will be invalid!")
             sample_var = np.zeros(y.shape)
 
+        # check freq_weight should be integer and should be accompanied by sample_var
+        if np.any(np.not_equal(np.mod(freq_weight, 1), 0)):
+            raise AttributeError("Frequency weights must all be integers for inference to be valid!")
         if sample_var.ndim < 2:
-            if np.any(np.equal(sample_weight, 1) & np.not_equal(sample_var, 0)):
+            if np.any(np.equal(freq_weight, 1) & np.not_equal(sample_var, 0)):
                 warnings.warn(
-                    "Variance was set to non-zero for an observation with sample_weight=1! "
+                    "Variance was set to non-zero for an observation with freq_weight=1! "
                     "sample_var represents the variance of the original observations that are "
                     "summarized in this sample. Hence, cannot have a non-zero variance if only "
                     "one observations was summarized. Inference will be invalid!")
+            elif np.any(np.not_equal(freq_weight, 1) & np.equal(sample_var, 0)):
```

Comment:
This seems wrong - isn't it possible that for some row, I observed all the same value (so sample variance is 0), even if for most rows that's not true?  Maybe this should be 
```suggestion
            elif np.all(np.not_equal(freq_weight, 1) & np.equal(sample_var, 0)):
```

Reply:
What do you mean? here is to check if all the variance is 0, but some freq_weight are not 1, then freq_weight is the same as sample_weight, similar with the checking above, check for the opposite case (if weights are all 1 but any of variance is not 0 )